### PR TITLE
Replace data providers and scrolling in new history

### PR DESCRIPTION
**Pull request by guerler**
This PR overhauls the beta history revision by replacing the rxjs-based data providers with a simple regular polling provider for now. It also strips the experimental scrolling logic and replaces it with a simple infinite scrolling module. This is the first iteration for the new history. In the next iteration we will redesign the components and come up with a new plan to cache results and reduce the query frequency. As of right now the query frequency is high with a throttling of only 500ms between requests. While this might not be optimal for production, it allows us to properly test and develop all required features before implementing a smarter query logic. The goal of this PR is to allow us to move forward with a clean slate.

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by bgruening** [comment]
Out of interest, are those 10.000 lines of code lost, or are we able to reuse them later?

File: client/src/components/History/CurrentCollection/Panel.vue [hunk 1]
```diff
@@ -1,97 +1,93 @@
...
+        };
+    },
+    created() {
+        this.dsc = new DatasetCollection(this.selectedCollection);
+    },
     computed: {
         selectedCollection() {
             const arr = this.selectedCollections;
             const selected = arr[arr.length - 1];
+            console.log(selected);
```
Comment (mvdbeek) [inline client/src/components/History/CurrentCollection/Panel.vue]:
```suggestion
```

File: client/src/components/History/model/SearchParams.js [hunk 2]
```diff
@@ -152,7 +103,7 @@ export class SearchParams {
                     // deleted serverField was reserved by pouchDB, needed to rename it to "isDeleted"
                     case "deleted":
                     case "visible":
-                        serverVal = dumbBool(val);
+                        serverVal = pythonBool(val);
```
Comment (mvdbeek) [inline client/src/components/History/model/SearchParams.js:106]: As much as I dislike we do it, that is a variable name that is 💯 times better

**Comment by guerler** [comment]
I honestly do think its accurate and fair. There are major performance issues and a considerable amount of bugs with the existing providers. They are not even handling the monitoring of an individual dataset properly when attempting to monitor changes. If these providers would be functional and well tested, this project would not have taken so long and the original developer would have completed it. Nevertheless it is possible that we can gain insights from this project and adopt some of its strategies, there is even a small chance that we will fix the issues and revive some of this code. Overall these providers are an experimental attempt to solve an interesting problem but are unnecessarily complicated and incomplete and the main driver of lost time.

File: client/src/components/History/readme.md [hunk 3]
```diff
@@ -1,69 +0,0 @@
-### Beta History Panel Component Tree
```
Comment (mvdbeek) [inline client/src/components/History/readme.md:1]: We should update that, even if it is subject to change.

File: client/src/components/WorkflowInvocationState/WorkflowInvocationStep.vue [hunk 4]
```diff
@@ -11,7 +11,8 @@
                 <InvocationStepProvider
                     v-if="isReady && invocationStepId !== undefined"
                     :id="invocationStepId"
-                    v-slot="{ item: stepDetails, loading }">
+                    auto-refresh="true"
```
Comment (mvdbeek) [inline client/src/components/WorkflowInvocationState/WorkflowInvocationStep.vue:14]: 😍

**Review by mvdbeek — COMMENTED** [review]
This looks good to me, and I think this is the right direction. My 2c about deleting the code ... we don't use it now, we should delete it. We have the git history if we need inspiration, but deleting dead code makes it a lot easier to focus on the important parts and for other developers to come in. That said, don't we have to clean up package.json ? Is there some tooling to automate that ?

File: lib/galaxy_test/selenium/test_history_panel_collections.py [hunk 5]
```diff
@@ -36,11 +36,8 @@ def test_mapping_collection_states_terminal(self):
         )
         ok_hid = ok_response["implicit_collections"][0]["hid"]
 
-        if not self.is_beta_history():
-            # sleep really shouldn't be needed :(
-            time.sleep(1)
-
-            self.home()
+        # sleep really shouldn't be needed :(
```
Comment (mvdbeek) [inline lib/galaxy_test/selenium/test_history_panel_collections.py]:
```suggestion
```

File: lib/galaxy_test/selenium/test_history_panel_collections.py [hunk 6]
```diff
@@ -222,12 +219,13 @@ def test_paired_display(self):
...
         if self.is_beta_history():
             self.sleep_for(WAIT_TYPES.HISTORY_POLL)
-            dataset_elements = self.content_item_by_attributes().all()
+            dataset_elements = collection_view.list_items_beta.all()
         else:
             dataset_elements = collection_view.list_items.all()
 
         assert len(dataset_elements) == 2, dataset_elements
         titles = [de.find_element_by_css_selector(".title .name").text for de in dataset_elements]
+        titles.sort()
```
Comment (mvdbeek) [inline lib/galaxy_test/selenium/test_history_panel_collections.py:227]: Shouldn't we sort that in the client ? (Which might as well be a bug we can fix separately)

Comment (guerler) [reply lib/galaxy_test/selenium/test_history_panel_collections.py:227]: Yes, this is a weird one. The order seems to be unstable i.e. this test passes locally but fails when run within the PR. I noted it and intend to revisit this once we have revised the rendering components for the collections.

**Comment by guerler** [comment]
That's a good point, looking at it now, I see the `pouchdb` and `rxjs` dependencies.

**Comment by mvdbeek** [comment]
Something seems wrong inside collections
Screenshot 2022-02-10 at 20 47 28
In the console I'm seeing
Screenshot 2022-02-10 at 20 49 49

And it seems to make requests to http://127.0.0.1:8000/api/histories/undefined/contents/undefineds/4fc821af39c7f4d5

**Comment by guerler** [comment]
I will work on a fix for this, such that it doesn't fail as dramatically. You are right, thanks.

**Comment by guerler** [comment]
I added a minor temporary fix, however as I mentioned that component will be rewritten soon and this part is going to change.

**Review by mvdbeek — APPROVED** [review]

**Comment by guerler** [comment]
Thank you so much for the review and comments.

**Pull request merged**